web/src/todo_widget.ts

```diff
@@ -440,33 +447,46 @@ export function activate({
 
         $elem.find("button.add-task").on("click", (e) => {
             e.stopPropagation();
-            $elem.find(".widget-error").text("");
+
             const task = $elem.find<HTMLInputElement>("input.add-task").val()?.trim() ?? "";
             const desc = $elem.find<HTMLInputElement>("input.add-desc").val()?.trim() ?? "";
 
-            if (task === "") {
-                return;
-            }
-
             $elem.find("input.add-task").val("").trigger("focus");
             $elem.find("input.add-desc").val("");
 
-            const task_exists = task_data.name_in_use(task);
-            if (task_exists) {
-                $elem.find(".widget-error").text($t({defaultMessage: "Task already exists"}));
-                return;
-            }
```

Comment:
Is the thinking here that we expect that the caller will have checked this before rendering the button? I am not sure we can delete this code -- at least, I'm not sure we've handled all case. Is it impossible to click before the button is disabled? If a task gets added by another client, does it live-update properly? 

We should test those scenarios, but I wonder if it'd be better to leave this code with just some comments noting those types of invalid input should usually not reach this point.